cmd/thanos/store.go

```diff
@@ -363,14 +363,21 @@ func runStore(
 
 			level.Info(logger).Log("msg", "initializing bucket store")
 			begin := time.Now()
-			if err := bs.InitialSync(ctx); err != nil {
+
+			err := runutil.RetryWithLog(logger, 2*time.Second, ctx.Done(), func() error {
```

Comment:
Can we make the interval configurable? Btw using `ctx.Done` here doesn't have any timeout. Let use another context with a configurable timeout.

Reply:
Yep that makes a lot of sense! Have added 2 flags, `--initial-sync-retry-timeout` and `--initial-sync-retry-interval` for configuring the retry behavior.

Reply:
I'm a bit wary of adding two new flags just for this feature, my initial idea was to have a simple retry mechanism :thinking:. I think having configurable interval makes sense, but how about setting some sensible timeout for all cases?

Reply:
I did a brief perusal of the codebase and found a `timeout` flag for downloading metadata from a bucket in [`tools_bucket.go`](https://github.com/thanos-io/thanos/blob/7ba274c69cb0a848c7e8e1c593221faa80d8b6e6/cmd/thanos/tools_bucket.go#L199). Perhaps we could still retain the `--initial-sync-retry-timeout` flag to allow user configuration of bucket timeout across components (if that makes sense, cmiiw), else we can set the timeout to 5 minutes for all cases following the default for `timeout` in `tools_bucket.go`

Reply:
I'm not strictly against, just a bit cautious about adding new flags, just to keep things simple for users (I guess initial sync is not such a defining feature that many users would need to tweak this know, but might be just my feeling). Thoughts @yeya24?

Either way let's try to move this forward, it looks good to me otherwise!

Reply:
ping @yeya24, looks like there's not much missing to carry this over the finish line.